src/operator/keycloak.ts

```diff
@@ -415,32 +375,36 @@ async function keycloakRealmProviderConfigurer(api: KeycloakApi) {
   realmConf.offlineSessionMaxLifespan = env.KEYCLOAK_TOKEN_OFFLINE_TTL
   // the api does not offer a list method, and trying to get by id throws an error
   // which we wan to discard, so we run the next command with an empty errors array
-  const existingRealm = (await doApiCall([], `Getting realm ${keycloakRealm}`, () =>
-    api.realms.realmGet(keycloakRealm),
-  )) as RealmRepresentation
-  if (existingRealm) {
-    await doApiCall(errors, `Updating realm ${keycloakRealm}`, async () =>
-      api.realms.realmPut(keycloakRealm, realmConf),
-    )
-  } else {
-    await doApiCall(errors, `Creating realm ${keycloakRealm}`, () => api.realms.rootPost(realmConf))
+  console.info(`Getting realm ${keycloakRealm}`)
+  try {
+    const existingRealm = (await api.realms.realmGet(keycloakRealm)).body
+    if (isUpdated(realmConf, existingRealm)) {
+      console.info(`Updating realm ${keycloakRealm}`)
+      await api.realms.realmPut(keycloakRealm, realmConf)
+    } else {
+      console.info(`Realm ${keycloakRealm} does not require updating`)
+    }
+  } catch (error) {
+    if (error.statusCode === 404) {
+      console.info(`Creating realm ${keycloakRealm}`)
+      await api.realms.rootPost(realmConf)
+    } else {
+      throw error
+    }
   }
 
   // Create Client Scopes
   const scope = createClientScopes()
-  const clientScopes = (await doApiCall(errors, 'Getting openid client scope', () =>
-    api.clientScope.realmClientScopesGet(keycloakRealm),
-  )) as Array<ClientScopeRepresentation>
+  console.info('Getting openid client scope')
+  const clientScopes = (await api.clientScope.realmClientScopesGet(keycloakRealm)).body as ClientScopeRepresentation[]
```

Comment:
Ah oke I think I did not understand the structure correctly. So when we get an error it will retry because of the retry operation. 
I was thinking maybe if you get an error back you still want to create the openid client scope or something. But that seems not to be the case.

Reply:
It would of course be better to check if all of the following requests are logically dependent on one another, but I have not looked into that sort of refactoring yet. The main objective for now was to increase the visibility of errors, so we can respond to them better, but we should look into some of these optimizations later.